### ISSUE TITLE
Connection from/to windows firefox clients not possible

### ISSUE THREAD

**Issue by wetsixteen** [nextcloud/spreed#800]
## Steps to reproduce
1. Login to cloud instance on firefox
2. Start video call / join call in firefox with other user
3. Answer call with other user, e.g. in chrome or iOS talk app

### Expected behaviour
Conference starts, video&audio gets transmitted
(Actually does exactly this when communication from/to linux-firefox, chrome, iOS app)

### Actual behaviour
Loading animation spins, no audio/video is transmitted

## Browser

**Microphone available:** yes

**Camera available:** yes

**Operating system:** Windows 7

**Browser name:** Firefox

**Browser version:** 59.0.2 (64bit)

### Browser log

<details>
```
Insert your browser log here, this could for example include:
a) The javascript console log
b) The network log
c) ...
```

</details>

## Spreed app

**Spreed app version:** 3.1.0
**Custom TURN server configured:** yes

**Custom STUN server configured:** yes

## Server configuration

**Operating system**: Linux 3.10.0-693.21.1.el7.x86_64 #1 SMP Wed Mar 7 19:03:37 UTC 2018 x86_64

**Web server:** Apache/2.4.6 (CentOS) OpenSSL/1.0.2k-fips SVN/1.7.14 PHP/7.1.16 (fpm-fcgi)

**Database:** mysql 5.5.56

**PHP version:** PHP 7.1.16
Modules loaded: Core, date, libxml, openssl, pcre, zlib, filter, hash, Reflection, SPL, session, standard, cgi-fcgi, bz2, calendar, ctype, curl, dom, mbstring, fileinfo, ftp, gd, gettext, iconv, json, ldap, exif, mysqlnd, PDO, apcu, posix, shmop, SimpleXML, sockets, sqlite3, sysvmsg, sysvsem, sysvshm, tokenizer, xml, xmlwriter, xsl, mysqli, pdo_mysql, pdo_sqlite, wddx, xmlreader, apc, igbinary, zip, Phar, redis, Zend OPcache

**Nextcloud Version:** 13.0.1 - 13.0.1.1

**List of activated apps:**

<details>
```
Enabled:
 - activity: 2.6.1
 - announcementcenter: 3.2.1
 - apporder: 0.4.1
 - bruteforcesettings: 1.0.3
 - calendar: 1.6.1
 - comments: 1.3.0
 - contacts: 2.1.3
 - dav: 1.4.6
 - federatedfilesharing: 1.3.1
 - federation: 1.3.0
 - files: 1.8.0
 - files_external: 1.4.1
 - files_markdown: 2.0.4
 - files_pdfviewer: 1.2.1
 - files_sharing: 1.5.0
 - files_texteditor: 2.5.1
 - files_trashbin: 1.3.0
 - files_versions: 1.6.0
 - files_videoplayer: 1.2.0
 - firstrunwizard: 2.2.1
 - gallery: 18.0.0
 - groupfolders: 1.2.0
 - issuetemplate: 0.3.0
 - logreader: 2.0.0
 - lookup_server_connector: 1.1.0
 - nextcloud_announcements: 1.2.0
 - notifications: 2.1.2
 - oauth2: 1.1.0
 - ojsxc: 3.3.2
 - password_policy: 1.3.0
 - provisioning_api: 1.3.0
 - radio: 0.6.1
 - rainloop: 5.0.6
 - richdocuments: 2.0.8
 - serverinfo: 1.3.0
 - sharebymail: 1.3.0
 - spreed: 3.1.0
 - survey_client: 1.1.0
 - systemtags: 1.3.0
 - tasks: 0.9.6
 - theming: 1.4.1
 - twofactor_backupcodes: 1.2.3
 - updatenotification: 1.3.0
 - user_ldap: 1.3.1
 - workflowengine: 1.3.0
Disabled:
 - admin_audit
 - encryption
 - keeweb
 - quota_warning
 - twofactor_totp
 - user_external
```
</details>

**Nextcloud configuration:**

<details>
```
{
    "instanceid": "***REMOVED SENSITIVE VALUE***",
    "passwordsalt": "***REMOVED SENSITIVE VALUE***",
    "secret": "***REMOVED SENSITIVE VALUE***",
    "trusted_domains": [
        "cloud.domain.tld"
    ],
    "datadirectory": "***REMOVED SENSITIVE VALUE***",
    "overwrite.cli.url": "https:\/\/cloud.domain.tld",
    "dbtype": "mysql",
    "version": "13.0.1.1",
    "dbname": "***REMOVED SENSITIVE VALUE***",
    "dbhost": "***REMOVED SENSITIVE VALUE***",
    "dbport": "",
    "dbtableprefix": "oc_",
    "dbuser": "***REMOVED SENSITIVE VALUE***",
    "dbpassword": "***REMOVED SENSITIVE VALUE***",
    "installed": true,
    "ldapUserCleanupInterval": 51,
    "versions_retention_obligation": "auto, 365",
    "memcache.local": "\\OC\\Memcache\\Redis",
    "redis": {
        "host": "***REMOVED SENSITIVE VALUE***",
        "port": 6379
    },
    "mail_domain": "***REMOVED SENSITIVE VALUE***",
    "mail_smtpmode": "php",
    "mail_smtpauthtype": "LOGIN",
    "mail_from_address": "***REMOVED SENSITIVE VALUE***",
    "maintenance": false,
    "loglevel": 2,
    "mail_smtphost": "***REMOVED SENSITIVE VALUE***",
    "mail_smtpport": "587",
    "htaccess.RewriteBase": "\/",
    "skeletondirectory": "\/var\/clouddata\/.skeleton\/",
    "mail_smtpsecure": "tls",
    "updater.release.channel": "stable"
}
```
</details>

### Server log (data/nextcloud.log)
nothing in serverlog related to this

edit: windows 10 & firefox works fine... (& fixed format)

**Comment by HAVP** [comment]
Did you get the Remote Candidates? Maybe we have the same problem.
Can you check this in Firefox with about:webrtc

**Comment by wetsixteen** [comment]
Many Remote Candidates (Externer Kandidat?) have ICE-Status 'failed'.

**Comment by HAVP** [comment]
That is ok. Only one ICE-Status with 'success' would be enough. So all connections has to fail when you get no connection. Do you see incomplete lines? You should also see connections to your turn server (local and remote candidate) which should never fail. But sometimes I also see this behavior.
Sorry, I can't help. I think the help of the developers is needed.

**Comment by nickvergessen** [comment]
You said it's working in one case, but not in the other.
Is there a difference? Like are you using different networks then?

If so, it might be the case, that you need a turn server, so make sure your turn server is configured correctly.

**Comment by wetsixteen** [comment]
turn server is correctly configured, and works with all tested browser combinations, except when win7+ firefox is involved.

no different networks involved when changing browser combinations.

**Comment by wetsixteen** [comment]
problem persists in 3.2

**Comment by nickvergessen** [comment]
Works quite fine here, so I still assume there is a problem with your Turn, Stun server or your internet connection

**Comment by wetsixteen** [comment]
Windows 7 + Firefax calls Mobile App:
<details>
`48: handle_udp_packet: New UDP endpoint: local addr <TURN NAT IP>:3478, remote addr <Mobile App IP>:12496
48: session 001000000000000001: realm <cloud.domain.tld> user <>: incoming packet BINDING processed, success
48: session 001000000000000001: realm <cloud.domain.tld> user <>: incoming packet message processed, error 401: Unauthorized
48: IPv4. Local relay addr: <TURN NAT IP>:62674
48: session 001000000000000001: new, realm=<cloud.domain.tld>, username=<1533378638>, lifetime=600
48: session 001000000000000001: realm <cloud.domain.tld> user <1533378638>: incoming packet ALLOCATE processed, success
48: IPv4. tcp or tls connected to: <Mobile App IP>:2656
48: session 001000000000000002: realm <cloud.domain.tld> user <>: incoming packet message processed, error 401: Unauthorized
48: IPv4. Local relay addr: <TURN NAT IP>:61135
48: session 001000000000000002: new, realm=<cloud.domain.tld>, username=<1533378638>, lifetime=600
48: session 001000000000000002: realm <cloud.domain.tld> user <1533378638>: incoming packet ALLOCATE processed, success
49: IPv4. tcp or tls connected to: <win7 external IP>:52487
51: session 001000000000000001: peer <Win 7 Internal IP> lifetime updated: 300
51: session 001000000000000001: realm <cloud.domain.tld> user <1533378638>: incoming packet CREATE_PERMISSION processed, success
51: session 001000000000000002: peer <Win 7 Internal IP> lifetime updated: 300
51: session 001000000000000002: realm <cloud.domain.tld> user <1533378638>: incoming packet CREATE_PERMISSION processed, success
58: session 001000000000000001: realm <cloud.domain.tld> user <1533378638>: incoming packet BINDING processed, success
68: session 001000000000000001: realm <cloud.domain.tld> user <1533378638>: incoming packet BINDING processed, success
69: session 001000000000000001: refreshed, realm=<cloud.domain.tld>, username=<1533378638>, lifetime=0
69: session 001000000000000001: realm <cloud.domain.tld> user <1533378638>: incoming packet REFRESH processed, success
69: session 001000000000000002: refreshed, realm=<cloud.domain.tld>, username=<1533378638>, lifetime=0
69: session 001000000000000002: realm <cloud.domain.tld> user <1533378638>: incoming packet REFRESH processed, success
69: session 001000000000000002: TCP socket closed remotely <Mobile App IP>:2656
69: session 001000000000000002: closed (2nd stage), user <1533378638> realm <cloud.domain.tld> origin <>, local <TURN NAT IP>:3478, remote <mobile ip>:2656, reason: TCP connection closed by client (callback)
69: session 001000000000000002: delete: realm=<cloud.domain.tld>, username=<1533378638>
69: session 001000000000000002: peer <Win 7 Internal IP> deleted
70: session 001000000000000001: closed (2nd stage), user <1533378638> realm <cloud.domain.tld> origin <>, local <TURN NAT IP>:3478, remote <mobile ip>:12496, reason: allocation timeout
70: session 001000000000000001: delete: realm=<cloud.domain.tld>, username=<1533378638>
70: session 001000000000000001: peer <Win 7 Internal IP> deleted`
</details>


Windows 7 + Chrome calls Mobile App:
<details>
`454: IPv4. tcp or tls connected to: <win7 external IP>:52645
454: IPv4. tcp or tls connected to: <win7 external IP>:52648
454: IPv4. tcp or tls connected to: <win7 external IP>:52651
454: session 001000000000000003: realm <cloud.domain.tld> user <>: incoming packet message processed, error 401: Unauthorized
454: session 001000000000000004: realm <cloud.domain.tld> user <>: incoming packet message processed, error 401: Unauthorized
454: session 001000000000000005: realm <cloud.domain.tld> user <>: incoming packet message processed, error 401: Unauthorized
454: IPv4. Local relay addr: <TURN NAT IP>:65275
454: session 001000000000000003: new, realm=<cloud.domain.tld>, username=<1533379030>, lifetime=600
454: session 001000000000000003: realm <cloud.domain.tld> user <1533379030>: incoming packet ALLOCATE processed, success
454: IPv4. Local relay addr: <TURN NAT IP>:62215
454: session 001000000000000005: new, realm=<cloud.domain.tld>, username=<1533379030>, lifetime=600
454: session 001000000000000005: realm <cloud.domain.tld> user <1533379030>: incoming packet ALLOCATE processed, success
454: IPv4. Local relay addr: <TURN NAT IP>:54055
454: session 001000000000000004: new, realm=<cloud.domain.tld>, username=<1533379030>, lifetime=600
454: session 001000000000000004: realm <cloud.domain.tld> user <1533379030>: incoming packet ALLOCATE processed, success
456: handle_udp_packet: New UDP endpoint: local addr <TURN NAT IP>:3478, remote addr <Mobile App IP>:5711
456: session 001000000000000006: realm <cloud.domain.tld> user <>: incoming packet BINDING processed, success
456: session 001000000000000006: realm <cloud.domain.tld> user <>: incoming packet message processed, error 401: Unauthorized
456: IPv4. Local relay addr: <TURN NAT IP>:59033
456: session 001000000000000006: new, realm=<cloud.domain.tld>, username=<1533379043>, lifetime=600
456: session 001000000000000006: realm <cloud.domain.tld> user <1533379043>: incoming packet ALLOCATE processed, success
456: IPv4. tcp or tls connected to: <Mobile App IP>:19579
456: session 000000000000000002: realm <cloud.domain.tld> user <>: incoming packet message processed, error 401: Unauthorized
456: session 001000000000000006: peer <Win 7 Internal IP> lifetime updated: 300
456: session 001000000000000006: realm <cloud.domain.tld> user <1533379043>: incoming packet CREATE_PERMISSION processed, success
456: session 001000000000000006: peer <TURN NAT IP> lifetime updated: 300
456: session 001000000000000006: realm <cloud.domain.tld> user <1533379043>: incoming packet CREATE_PERMISSION processed, success
456: IPv4. Local relay addr: <TURN NAT IP>:64211
456: session 000000000000000002: new, realm=<cloud.domain.tld>, username=<1533379043>, lifetime=600
456: session 000000000000000002: realm <cloud.domain.tld> user <1533379043>: incoming packet ALLOCATE processed, success
456: session 000000000000000002: peer <Win 7 Internal IP> lifetime updated: 300
456: session 000000000000000002: realm <cloud.domain.tld> user <1533379043>: incoming packet CREATE_PERMISSION processed, success
456: session 000000000000000002: peer <TURN NAT IP> lifetime updated: 300
456: session 000000000000000002: realm <cloud.domain.tld> user <1533379043>: incoming packet CREATE_PERMISSION processed, success
457: session 001000000000000004: refreshed, realm=<cloud.domain.tld>, username=<1533379030>, lifetime=0
457: session 001000000000000004: realm <cloud.domain.tld> user <1533379030>: incoming packet REFRESH processed, success
457: session 001000000000000005: refreshed, realm=<cloud.domain.tld>, username=<1533379030>, lifetime=0
457: session 001000000000000005: realm <cloud.domain.tld> user <1533379030>: incoming packet REFRESH processed, success
457: session 001000000000000004: TCP socket closed remotely <win7 external IP>:52648
457: session 001000000000000004: closed (2nd stage), user <1533379030> realm <cloud.domain.tld> origin <>, local <TURN NAT IP>:3478, remote <win7 external IP>:52648, reason: TCP connection closed by client (callback)
457: session 001000000000000004: delete: realm=<cloud.domain.tld>, username=<1533379030>
457: session 001000000000000005: TCP socket closed remotely <win7 external IP>:52651
457: session 001000000000000005: closed (2nd stage), user <1533379030> realm <cloud.domain.tld> origin <>, local <TURN NAT IP>:3478, remote <win7 external IP>:52651, reason: TCP connection closed by client (callback)
457: session 001000000000000005: delete: realm=<cloud.domain.tld>, username=<1533379030>
458: session 001000000000000003: peer <Mobile App IP> lifetime updated: 300
458: session 001000000000000003: realm <cloud.domain.tld> user <1533379030>: incoming packet CREATE_PERMISSION processed, success
458: session 001000000000000003: peer <TURN NAT IP> lifetime updated: 300
458: session 001000000000000003: realm <cloud.domain.tld> user <1533379030>: incoming packet CREATE_PERMISSION processed, success
458: session 001000000000000003: peer <TURN NAT IP> lifetime updated: 300
458: session 001000000000000003: realm <cloud.domain.tld> user <1533379030>: incoming packet CREATE_PERMISSION processed, success
458: session 001000000000000006: peer <TURN NAT IP> lifetime updated: 600
458: session 001000000000000006: realm <cloud.domain.tld> user <1533379043>: incoming packet CHANNEL_BIND processed, success
458: session 001000000000000003: peer <TURN NAT IP> lifetime updated: 600
458: session 001000000000000003: realm <cloud.domain.tld> user <1533379030>: incoming packet CHANNEL_BIND processed, success
464: session 001000000000000006: usage: realm=<cloud.domain.tld>, username=<1533379043>, rp=1498, rb=1198597, sp=550, sb=42322
464: session 001000000000000003: usage: realm=<cloud.domain.tld>, username=<1533379030>, rp=559, rb=43511, sp=1489, sb=1202968
465: session 001000000000000006: refreshed, realm=<cloud.domain.tld>, username=<1533379043>, lifetime=0
465: session 001000000000000006: realm <cloud.domain.tld> user <1533379043>: incoming packet REFRESH processed, success
465: session 000000000000000002: refreshed, realm=<cloud.domain.tld>, username=<1533379043>, lifetime=0
465: session 000000000000000002: realm <cloud.domain.tld> user <1533379043>: incoming packet REFRESH processed, success
465: session 000000000000000002: TCP socket closed remotely <Mobile App IP>:19579
465: session 000000000000000002: closed (2nd stage), user <1533379043> realm <cloud.domain.tld> origin <>, local <TURN NAT IP>:3478, remote <Mobile App IP>:19579, reason: TCP connection closed by client (callback)
465: session 000000000000000002: delete: realm=<cloud.domain.tld>, username=<1533379043>
465: session 000000000000000002: peer <Win 7 Internal IP> deleted
465: session 000000000000000002: peer <TURN NAT IP> deleted
465: session 001000000000000003: refreshed, realm=<cloud.domain.tld>, username=<1533379030>, lifetime=0
465: session 001000000000000003: realm <cloud.domain.tld> user <1533379030>: incoming packet REFRESH processed, success
465: session 001000000000000003: TCP socket closed remotely <win7 external IP>:52645
465: session 001000000000000003: closed (2nd stage), user <1533379030> realm <cloud.domain.tld> origin <>, local <TURN NAT IP>:3478, remote <win7 external IP>:52645, reason: TCP connection closed by client (callback)
465: session 001000000000000003: delete: realm=<cloud.domain.tld>, username=<1533379030>
465: session 001000000000000003: peer <TURN NAT IP> deleted
465: session 001000000000000003: peer <Mobile App IP> deleted
466: session 001000000000000006: closed (2nd stage), user <1533379043> realm <cloud.domain.tld> origin <>, local <TURN NAT IP>:3478, remote <Mobile App IP>:5711, reason: allocation timeout
466: session 001000000000000006: delete: realm=<cloud.domain.tld>, username=<1533379043>
466: session 001000000000000006: peer <Win 7 Internal IP> deleted
466: session 001000000000000006: peer <TURN NAT IP> deleted`
</details>

**Comment by misi** [comment]
@wetsixteen  @nickvergessen
I have just tested coturn + FF on linux and I don't find any issue... :(
